test/test_interpop.py

```diff
@@ -0,0 +1,152 @@
+#   notice, this list of conditions and the following disclaimer in the
+#   documentation and/or other materials provided with the distribution.
+# * Neither the name of The Bifrost Authors nor the names of its
+#   contributors may be used to endorse or promote products derived
+#   from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+import numpy as np
+import bifrost as bf
+
+from bifrost.libbifrost_generated import BF_CUDA_ENABLED
+
+try:
+    import cupy as cp
+    HAVE_CUPY = True
+except ImportError:
+    HAVE_CUPY = False
+
+try:
+    import pycuda.driver as cuda
+    import pycuda.autoinit
+    import pycuda.gpuarray
+    import pycuda.driver
+    HAVE_PYCUDA = True
+except ImportError:
+    HAVE_PYCUDA = False
+
+class BifrostStreamManager(object):
+    def __init__(self, stream):
+        self._stream = stream
+        
+    def __enter__(self):
+        self._orig_stream = bf.device.get_stream()
+        stream = getattr(self._stream, 'ptr', None)
+        if stream is None:
+            stream = getattr(self._stream, 'handle', None)
+        if stream is None:
+            stream = self._stream
+        bf.device.set_stream(stream)
+        return self
+        
+    def __exit__(self, type, value, tb):
+        bf.device.set_stream(self._orig_stream)
+
+@unittest.skipUnless(BF_CUDA_ENABLED and HAVE_CUPY, "requires GPU support and cupy")
+class TestCuPy(unittest.TestCase):
+    @staticmethod
+    def create_data():
+        data = np.random.rand(100,1000)
+        return data.astype(np.float32)
+        
+    def test_as_cupy(self):
+        data = self.create_data()
+        
+        bf_data = bf.ndarray(data, space='cuda')
+        cp_data = bf_data.as_cupy()
+        np_data = cp.asnumpy(cp_data)
+        np.testing.assert_allclose(np_data, data)
+    def test_from_cupy(self):
+        data = self.create_data()
+        
+        cp_data = cp.asarray(data)
+        bf_data = bf.ndarray(cp_data)
+        np_data = bf_data.copy(space='system')
+        np.testing.assert_allclose(np_data, data)
+        
+    def test_stream(self):
+        data = self.create_data()
+        
+        orig_stream = bf.device.get_stream()
```

Comment:
Curious why `orig_stream` is not used in the rest of this method. Is this call to `get_stream` redundant with (maybe leftover from before adding) `BifrostStreamManager`? Also happens on line 142.

Reply:
Yeah, looks like it is a remnant of the pre-BifrostStreamManager version of the test.